src/App.jsx

```diff
@@ -14,70 +15,65 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [checked, setChecked] = useState(8);
+  const [currentName, setCurrentName] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+  return (
+    <main className="section container">
+      {checked === null
+        ? (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )
+        : (
+          <h1 className="title is-flex is-align-items-center">
+            {`${currentName} is selected`}
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={() => setChecked(null)}
+            />
+          </h1>
+        )
+      }
+      <table className="table">
+        <tbody>
+          {goods.map((goodName, index) => {
```

Comment:
[CHECKLIST] 2.DON'T use index as a key.

Using array index as a key can cause unpredictable behavior. 

To avoid it you can map your original array to array of objects where name property is going to be the name of your good and id property will be the index of the current position in array. 
The difference is when you reorder elements in array, value will be moved to an array's cell with different index, at the same time an object's properties will always stay connected, no matter the order of elements.